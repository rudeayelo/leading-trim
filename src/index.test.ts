import { leadingTrim } from ".";

it("outputs styles correctly", () => {
  const leadingTrimCSS = leadingTrim({
    lineHeight: 1.5,
    reference: {
      topCrop: 7,
      bottomCrop: 7,
      fontSize: 40,
      lineHeight: 1.1,
    },
    correction: {
      top: 1,
      bottom: -1,
    },
  });

  expect(leadingTrimCSS).toMatchSnapshot();
});
