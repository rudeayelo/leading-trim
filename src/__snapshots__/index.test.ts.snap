// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs styles correctly 1`] = `
{
  "&::after": {
    "marginTop": "calc(-0.37499999999999994em - 1px + -1px)",
  },
  "&::before": {
    "marginBottom": "calc(-0.37499999999999994em - 1px + 1px)",
  },
  "&::before, &::after": {
    "content": """",
    "display": "block",
    "height": 0,
    "paddingTop": "1px",
    "width": 0,
  },
  "display": "block",
  "lineHeight": 1.5,
}
`;
